crates/nu-protocol/src/engine/stack.rs

```diff
@@ -36,6 +37,29 @@ pub struct Stack {
     /// List of active overlays
     pub active_overlays: Vec<String>,
     pub recursion_count: u64,
+
+    pub parent_stack: Option<Arc<Stack>>,
+    pub parent_deletions: Vec<VarId>,
+}
+
+impl Clone for Stack {
+    fn clone(&self) -> Self {
+        // uncomment this code to see if we are cloning the stack in the REPL instead of
+        // creating a child stack
+        //
+        // if (self.parent_stack.is_none()) {
+        //     panic!("Cloing a stack without a parent. Are we cloning the base stack?");
+        // }
```

Comment:
This was to help me find where I might have been unintentionally cloning the top-level stack for the REPL